fix(buondua): enhance Cloudflare challenge bypass

marioplus opened the pull request:
- Add rate limiting (max 10 requests/second)
- Implement random User-Agent rotation
- Inject Referer header
- Version bump 2 → 3

closes: #8079

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [ ] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [ ] Have removed `web_hi_res_512.png` when adding a new extension

marioplus committed:
fix(buondua): enhance Cloudflare challenge bypass

- Add rate limiting (max 10 requests/second)
- Implement random User-Agent rotation
- Inject Referer header
- Version bump 2 → 3

refs: #8079

choppeh commented on src/all/buondua/src/eu/kanade/tachiyomi/extension/all/buondua/BuonDua.kt:
```diff
@@ -1,25 +1,41 @@
+                .header("Referer", "$baseUrl/")
+                .build()
+            chain.proceed(request)
+        }
```
Replace the interceptor with ``` headersBuilder()```

```suggestion
```

https://github.com/keiyoushi/extensions-source/blob/228b657e25788f904b29cde7561b69d5a38f800b/lib-multisrc/goda/src/eu/kanade/tachiyomi/multisrc/goda/GoDa.kt#L30

choppeh commented on src/all/buondua/src/eu/kanade/tachiyomi/extension/all/buondua/BuonDua.kt:
```diff
@@ -77,7 +95,9 @@ class BuonDua() : ParsedHttpSource() {
-        chapter.date_upload = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US).parse(element.select(".article-info > small").text())?.time ?: 0L
+        chapter.date_upload =
+            SimpleDateFormat("H:m DD-MM-yyyy", Locale.US).parse(element.select(".article-info > small").text())?.time
+                ?: 0L
```
Move the ```SimpleDateFormat``` to the ```companion object``` or class variable: 
```kt
companion object {
     private val dateFormat = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US)
}
// or
private val dateFormat = SimpleDateFormat("H:m DD-MM-yyyy", Locale.US)
``` 

And use the ```keiyoushi.utils.tryParse```  from utils
```suggestion
        chapter.date_upload = dateFormat.tryParse(element.select(".article-info > small").text())
```

You need to import:
```kt 
import keiyoushi.utils.tryParse
```

https://github.com/keiyoushi/extensions-source/blob/228b657e25788f904b29cde7561b69d5a38f800b/lib-multisrc/mangabox/src/eu/kanade/tachiyomi/multisrc/mangabox/MangaBox.kt#L314

marioplus committed:
chore(buondua): replace interceptor with headersBuilder for request headers

- Replace the interceptor with  headersBuilder()
- Move the SimpleDateFormat to the companion object or class variable

Refs: #8079
